### PR TITLE
Connection: Timeout config should be independent between client instances

### PR DESCRIPTION
**Pull request by mckelvin**
The issue can be reproduced using `tests/shabby/independent_timeout.py`.

The same timeout **poll timeout** and **connect timeout** were shared by multi client instances in single process, which is not up to expectation.

cc @windreamer @zzl0

### CODE REVIEW

**Comment by xtao** [comment]
:clap:

**Comment by zzl0** [comment]
:beer:

File: include/Connection.h [hunk 1]
```diff
@@ -51,9 +51,9 @@ class Connection {
 
 
     void reset();
-    static void setRetryTimeout(int timeout);
-    static const int getRetryTimeout();
-    static void setConnectTimeout(int timeout);
+    void setRetryTimeout(int timeout);
```
Comment (zzl0) [inline include/Connection.h]: Can we use another name for this, `setRetryInterval` for example ?

Comment (mckelvin) [inline include/Connection.h]: :+1:  `Interval` is more correct. The old `Timeout`  is followed after`MEMCACHED_BEHAVIOR_RETRY_TIMEOUT` in libmemcached.

This is a incompatible API change, I'll consider rename it in next large MINOR version but not this time.

Looks good,

**Pull request merged**